io/io/src/TDirectoryFile.cxx

```diff
@@ -1214,7 +1211,7 @@ void TDirectoryFile::ls(Option_t *option) const
       for (TObjLink *lnk = fKeys->FirstLink(); lnk != nullptr; lnk = lnk->Next()) {
          TKey *key = (TKey*)lnk->GetObject();
          TString s = key->GetName();
-         if (user_reg && s.Index(re) == kNPOS)
+         if (!reg.IsNull() && s.Index(re) == kNPOS)
```

Comment:
The same check needs to be done on line 1204 (which was also missing an update in the previous incarnation).

Reply:
good catch!! fixed now